src/main/java/net/minecraftforge/common/crafting/CraftingHelper.java

```diff
@@ -582,20 +584,17 @@ public static void loadRecipes()
         //TODO: Figure out how to remove recipes, and override them. This relies on cpw to help.
         //For now this is only done one after mod init, I want to move this to ServerInit and re-do it many times.
         init();
+        ForgeRegistry<IRecipe> reg = (ForgeRegistry<IRecipe>)ForgeRegistries.RECIPES;
         if (DEBUG_LOAD_MINECRAFT)
-        {
-            Iterator<IRecipe> itr = GameData.getRecipeRegistry().iterator();
-            while(itr.hasNext())
-            {
-                itr.next();
-                itr.remove();
-            }
-        }
+            reg.clear();
+        else
+            GameData.revert(RegistryManager.FROZEN, GameData.RECIPES, false);
```

Comment:
won't this cause duplicate recipe registrations after reverting from frozen state? the registry is frozen after all moded recipes are loaded but the loading of those below isn't behind any checks so they would be loaded a 2nd time while already being in the registry